backend/dal/Services/Admin/Concrete/UserService.cs

```diff
@@ -150,6 +150,65 @@ public override void Remove(User entity)
             base.Remove(user);
         }
 
+        /// <summary>
+        /// Updates the specified access request in the datasource. if the request is granted, update the associated user as well.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public AccessRequest Update(AccessRequest entity)
+        {
+            if(entity.UserId == null)
+            {
+                throw new ArgumentNullException("userId");
+            }
+            entity.ThrowIfNull(nameof(entity));
+            this.User.ThrowIfNotAuthorized(Permissions.SystemAdmin, Permissions.AgencyAdmin);
+
+            var accessRequest = this.Context.AccessRequests
+                .Include(p => p.Agencies)
+                .ThenInclude(p => p.Agency)
+                .Include(p => p.Roles)
+                .ThenInclude(p => p.Role)
+                .Include(p => p.User)
+                .FirstOrDefault(a => a.Id == entity.Id) ?? throw new KeyNotFoundException();
+            if(!accessRequest.IsGranted != true && entity.IsGranted == true && accessRequest.UserId.HasValue)
```

Comment:
Any reason for the double negative?  `accessRequest.IsGranted == true`.  No reason to double check if `accessRequest.UserId.HasValue` as you've checked earlier already.

Reply:
I originally wasn't enforcing the non-null user, missed updating the trickle down logic after I started enforcing it.